graph/load.go

```diff
@@ -91,25 +86,79 @@ func (s *TagStore) CmdLoad(job *engine.Job) engine.Status {
 	} else if !os.IsNotExist(err) {
 		return job.Error(err)
 	}
-
 	return engine.StatusOK
 }
 
-func (s *TagStore) recursiveLoad(eng *engine.Engine, address, tmpImageDir string) error {
+func (s *TagStore) recursiveLoad(eng *engine.Engine, address string, tr *tar.Reader, repoTarFile string, flag bool) error {
 	if err := eng.Job("image_get", address).Run(); err != nil {
 		log.Debugf("Loading %s", address)
 
-		imageJson, err := ioutil.ReadFile(path.Join(tmpImageDir, "repo", address, "json"))
-		if err != nil {
-			log.Debugf("Error reading json", err)
-			return err
+		var trParent *tar.Reader
+		var imageJson []byte
+		var layer io.Reader
+		var hdr *tar.Header
+		var tarFile *os.File
+
+		if flag == false {
+			if tarFile, err = os.Open(repoTarFile); err != nil {
+				return err
+			}
+			trParent = tar.NewReader(tarFile)
+			for {
+				hdr, err = trParent.Next()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return err
+				}
+				if strings.Trim(hdr.Name, "/") == address {
+					break
+				}
+			}
 		}
 
-		layer, err := os.Open(path.Join(tmpImageDir, "repo", address, "layer.tar"))
-		if err != nil {
-			log.Debugf("Error reading embedded tar", err)
-			return err
+		for i := 0; i < 3; i++ {
```

Comment:
because there are 3 entries in each image layer. And I only need two of them {JSON, layer.tar}